### ISSUE TITLE
Issue type '{name}' appears more than once on `create_issue`

### ISSUE THREAD

**Issue by dblanchette** [pycontribs/jira#1444]
### Bug summary

Updating from version 3.0 to 3.3.1, we started seeing `Issue type 'User Story' appears more than once` when calling `create_issue` with a project and `User Story` issue type.

Example call:
```python
jira.create_issue(
    fields={
        "project": "PROJECTKEY",
        "issuetype": "User Story",
        "summary": "test",
    }
)
```

Looking at the code, a call is made to https://myorg.atlassian.net/rest/api/3/issuetype to retrieve all issue types. A solution would be to call https://myorg.atlassian.net/rest/api/3/project/PROJECTKEY/. I assume a project key or ID is provided since we cannot  create an issue outside of a project.

Something else I tried is:
```python
jira.create_issue(
    fields={
        "project": "PROJECTKEY",
        "issuetype": 2,
        "summary": "test",
    }
)
```
but this fails with `Issue type '2' is unknown.`

This is probably caused by [this code in client.py](https://github.com/pycontribs/jira/blob/1f0a2ae74700f990778fb698f43953f1ad5aa4a7/jira/client.py#L1502)
```python
p = data["fields"]["issuetype"]
if isinstance(p, int):
    data["fields"]["issuetype"] = {"id": p}
if isinstance(p, (str, int)):
    data["fields"]["issuetype"] = {"id": self.issue_type_by_name(str(p)).id}
```

It seems to me that if we provide a int, e.g. 2, then we first change it to `{"id": 2}` and then still try to find an issue type named "2" which fails.

Is there a specific reason why that would be the case? Maybe we want to handle an issue type being named a number? (But then, users should still provide a str).

Lastly, a workaround is to it call like so (if you know the issue type id):
```python
jira.create_issue(
    fields={
        "project": "PROJECTKEY",
        "issuetype": {"id": 2},
        "summary": "test",
    }
)
```

One can also list the issue types themself:
```python
project = jira_client.client.project("PROJECTKEY", expand="issueTypes")
issue_type_id = next(issue_type.id for issue_type in mun_project.issueTypes if issue_type.name == "User Story")
issue = jira_client.client.create_issue(
    fields={
        "project": "PROJECTKEY",
        "issuetype": {"id": issue_type_id},
        "summary": "test",
    }
)
```

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Jira Instance type

Jira Cloud (Hosted by Atlassian)

### jira-python version

3.3.1

### Python Interpreter version

3.8

### Which operating systems have you used?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Reproduction steps

```python
# 0. Given a Jira instance with multiple projects with an issue type named `User Story`, but only one in the project `PROJECTKEY`
# 1. Given a Jira client instance
jira: JIRA
# 2. When I call the function with argument x
issue = jira.create_issue(
    fields={
        "project": "PROJECTKEY",
        "issuetype": "User Story",
        "summary": "test",
    }
)
```

### Stack trace

```python
Traceback (most recent call last):
   File "test_jira.py", line 37, in test_jira_issue
     issue = jira_client.client.create_issue(
   File "/root/.cache/pypoetry/virtualenvs/something-BtA_kvLS-py3.8/lib/python3.8/site-packages/jira/client.py", line 1509, in create_issue
     data["fields"]["issuetype"] = {"id": self.issue_type_by_name(str(p)).id}
   File "/root/.cache/pypoetry/virtualenvs/something-BtA_kvLS-py3.8/lib/python3.8/site-packages/jira/client.py", line 2533, in issue_type_by_name
     raise KeyError(f"Issue type '{name}' appears more than once.")

KeyError: "Issue type 'User Story' appears more than once."
```

### Expected behaviour

`User Story` is looked up in the project `PROJECTKEY` and is unique

**Comment by adehad** [comment]
Thanks for the excellent report, we recently merged in #1445 that fixes this, would be great if you can confirm when installing the latest main `pip install git+https://github.com/pycontribs/jira.git@main`

For now closing

**Comment by dblanchette** [comment]
Thanks for the fix!

I confirm #1445 is working, but there is also the part of providing a string when the name of the issue type matches several issue types, but there is only one in the provided project, which I attempt to address in #1448.
